pkg/api/api.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+type ReportingStatus int
+
+const (
+	Started ReportingStatus = iota
+	Succeeded
+	Failed
+)
+
+// Reporter is responsible for reporting back on the progress of the cloud preparation
+type Reporter interface {
+	// Started will report that an operation started on the cloud
+	Started(message string, args ...interface{})
+
+	// Succeeded will report that the last operation on the cloud has succeded
+	Succeeded(message string, args ...interface{})
+
+	// Failed will report that the last operation on the cloud has failed
+	Failed(err ...error)
```

Comment:
An alternative to this could be using channels (Accepting messages for Started,Succeeded, Failed) ? 

That would be more go-native, and easier to consume on the caller.

Reply:
We could also keep this and provide a "channel reporter" if we think that would be useful.